Bottom: 1777c7f57e90500526fb43afa6e60b258c66c932
Top:    53b86e28e3a9ec9d020b57200daee901570670a9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 17:59:56 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6264a6f..eacb418 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1123,7 +1123,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
